src/badguy/dispenser.cpp

```diff
@@ -260,31 +257,27 @@ Dispenser::active_update(float dt_sec)
 void
 Dispenser::launch_badguy()
 {
-  if (m_badguys.empty()) return;
-  if (m_frozen) return;
-  if (m_limit_dispensed_badguys &&
-      m_current_badguys >= m_max_concurrent_badguys)
-      return;
+  if (m_badguys.empty() || m_frozen || (m_limit_dispensed_badguys && m_current_badguys >= m_max_concurrent_badguys)) {
+    return;
+  }
```

Comment:
Good compression, I'd recommend making a seperate line for that last condition combo given the length of the line.

And :b: :stuck_out_tongue: